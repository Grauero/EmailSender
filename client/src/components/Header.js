import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import Payments from './Payments';

const Header = ({ auth }) => {
  function renderContent() {
    switch (auth) {
      case null:
        return;
      case false:
        return (
          <li>
            <a href={process.env.REACT_APP_LOGIN}>Login With Google</a>
          </li>
        );
      default:
        return [
          <li key="1">
            <Payments />
          </li>,
          <li key="2" style={{ margin: '0 10px' }}>
            Credits: {auth.credits}
          </li>,
          <li key="3">
            <a href={process.env.REACT_APP_LOGOUT}>Logout</a>
          </li>
        ];
    }
  }

  return (
    <nav>
      <div className="nav-wrapper">
        <Link
          to={auth ? '/surveys' : '/'}
          className="left brand-logo"
          style={{ marginLeft: '10px' }}
        >
          <i className="material-icons left">email</i>EmailSender
        </Link>
        <ul className="right">{renderContent()}</ul>
      </div>
    </nav>
  );
};

Header.propTypes = {
  auth: PropTypes.oneOfType([PropTypes.instanceOf(Object), PropTypes.bool])
};

Header.defaultProps = {
  auth: null
};

const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(mapStateToProps)(Header);
