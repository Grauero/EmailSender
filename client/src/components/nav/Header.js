import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import Payments from '../utils/Payments';

function renderContent(auth) {
  switch (auth) {
    case null:
      return;
    case false:
      return (
        <li>
          <a href="/auth/google">Login With Google</a>
        </li>
      );
    default:
      return [
        <li key="1">
          <Payments />
        </li>,
        <li key="2" style={{ margin: '0 10px' }}>
          Credits: {auth.credits}
        </li>,
        <li key="3">
          <a href="/auth/logout">Logout</a>
        </li>
      ];
  }
}

const Header = ({ auth }) => (
  <nav>
    <div className="nav-wrapper">
      <Link
        to={auth ? '/surveys' : '/'}
        className="left brand-logo"
        style={{ marginLeft: '10px' }}
      >
        <i className="material-icons left">email</i>EmailSender
      </Link>
      <ul className="right">{renderContent(auth)}</ul>
    </div>
  </nav>
);

Header.propTypes = {
  auth: PropTypes.oneOfType([PropTypes.instanceOf(Object), PropTypes.bool])
};

Header.defaultProps = {
  auth: null
};

const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(mapStateToProps)(Header);
