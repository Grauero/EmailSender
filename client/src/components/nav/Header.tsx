import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import Payments from '../utils/Payments';
import { IAppState } from '../../store/reducers';
import { AuthState } from '../../store/reducers/authReducer';

function renderContent(auth: AuthState) {
  if (auth instanceof Object) {
    return [
      <li key="1">
        <Payments />
      </li>,
      <li key="2" style={{ margin: '0 10px' }}>
        Credits: {auth.credits}
      </li>,
      <li key="3">
        <a href="/auth/logout">Logout</a>
      </li>
    ];
  } else if (!auth) {
    return (
      <li>
        <a href="/auth/google">Login With Google</a>
      </li>
    );
  }

  return;
}

export interface IHeader {
  auth: AuthState;
}

const Header: React.FC<IHeader> = props => {
  const { auth } = props;

  return (
    <nav>
      <div className="nav-wrapper">
        <Link
          to={auth ? '/surveys' : '/'}
          className="left brand-logo"
          style={{ marginLeft: '10px' }}
        >
          <i className="material-icons left">email</i>EmailSender
        </Link>
        <ul className="right">{renderContent(auth)}</ul>
      </div>
    </nav>
  );
};

const mapStateToProps = (state: IAppState) => ({
  auth: state.auth
});

export default connect(mapStateToProps)(Header);
