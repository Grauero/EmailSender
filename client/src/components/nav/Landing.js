import React from 'react';

const Landing = () => {
  const style = { textAlign: 'center' };

  return (
    <div style={style}>
      <h1>EmailSender</h1>
      Collect feedback from your users
    </div>
  );
};

export default Landing;
