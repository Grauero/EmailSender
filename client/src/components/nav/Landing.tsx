import React from 'react';

const Landing: React.FC<{}> = () => (
  <div className="landing">
    <h1>EmailSender</h1>
    <img src="mail.png" alt="app logo" width="300px" />
    <p>Collect feedback from your users</p>
  </div>
);

export default Landing;
