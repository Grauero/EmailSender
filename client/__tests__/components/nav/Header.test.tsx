import 'jest';
import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { Header, mapStateToProps } from '../../../src/components/nav/Header';

test('renders logo', () => {
  const logo = shallow(<Header auth={false} />).find('Link');

  expect(toJSON(logo)).toMatchSnapshot();
});

test('renders Login button if user isnt authorized', () => {
  const nav = shallow(<Header auth={false} />).find('ul');

  expect(toJSON(nav)).toMatchSnapshot();
});

test('renders Credits and Logout button if user is authorized', () => {
  const props = { googleId: 'id', credits: 5 };
  const nav = shallow(<Header auth={props} />).find('ul');

  expect(toJSON(nav)).toMatchSnapshot();
});

test('maps auth value from store to props', () => {
  const props = mapStateToProps({ auth: false, surveys: [], form: null });

  expect(props).toEqual({ auth: false });
});
