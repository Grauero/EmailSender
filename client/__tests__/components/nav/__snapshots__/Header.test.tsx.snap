// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Credits and Logout button if user is authorized 1`] = `
<ul>
  <li
    key="1"
  >
    <Connect(Payments) />
  </li>
  <li
    key="2"
    style={
      Object {
        "margin": "0 10px",
      }
    }
  >
    Credits: 
    5
  </li>
  <li
    key="3"
  >
    <a
      href="/auth/logout"
    >
      Logout
    </a>
  </li>
</ul>
`;

exports[`renders Login button if user isnt authorized 1`] = `
<ul>
  <li>
    <a
      className="auth"
      href="/auth/google"
    >
      Login With Google
    </a>
  </li>
</ul>
`;

exports[`renders logo 1`] = `
<Link
  className="logo"
  replace={false}
  to="/"
>
  <i
    className="material-icons left"
  >
    email
  </i>
  <h2>
    EmailSender
  </h2>
</Link>
`;
